public/utils/datasource-utils.ts

```diff
@@ -30,7 +27,8 @@ export function getClusterInfo(dataSourceEnabled: boolean, cluster: DataSourceOp
 
 export function getDataSourceFromUrl(): DataSourceOption {
```

Comment:
Should this fn ever be called if MDS is disabled?

Should there be a safeguard in this fn to return something false-y if MDS is disabled?

Reply:
Blocked it behind the check to see if its enabled. Benefit of this approach is that it also allows reading the default datasource from the picker if available (when MDS is enabled).